### DISCUSSION TITLE
ItemTransaction

### DISCUSSION THREAD

**Discussion by adapap** [StevensSEC/pokemonbattlelib#241] [General]
Should we change `ItemTransaction` to have `Target target` instead of `Target *Pokemon`?

Rationale:

[Black Sludge](https://veekun.com/dex/items/misc/black%20sludge) creates `DamageTransaction` which requires `target`. We don't pass in party/slot into an `ItemTransaction`.

```go
case ItemBlackSludge:
		if t.Target.Type&TypePoison != 0 {
			b.QueueTransaction(HealTransaction{
				Target: t.Target,
				Amount: t.Target.MaxHP() / 16,
			})
		} else {
			// Need target, change ItemTransaction.Target from *Pokemon to target?
			// b.QueueTransaction(DamageTransaction{
			// 	Target: t,
			// 	Damage: t.Target.MaxHP() / 8,
			// })
		}
```

**Comment by adapap** [comment]
@dyc3

**Reply by dyc3** [reply]
Yes, we totally should. I'll convert this into an issue.
